environs/tools/simplestreams.go

```diff
@@ -167,24 +173,29 @@ func Fetch(
 		StreamsVersion:   currentStreamsVersion,
 		LookupConstraint: cons,
 		ValueParams: simplestreams.ValueParams{
-			DataType:        ContentDownload,
-			FilterFunc:      appendMatchingTools,
-			MirrorContentId: ToolsContentId(cons.Stream),
-			ValueTemplate:   ToolsMetadata{},
+			DataType:      ContentDownload,
+			FilterFunc:    appendMatchingTools,
+			ValueTemplate: ToolsMetadata{},
 		},
 	}
-	items, resolveInfo, err := simplestreams.GetMetadata(sources, params)
+	if len(cons.Streams) > 0 {
+		params.ValueParams.MirrorContentId = ToolsContentId(cons.Streams[0])
```

Comment:
I'm a bit unsure about this part. It feels odd to rely on the knowledge that the first stream will be the one we're officially on.

Reply:
It's not - the there will potentially be multiple mirror content ids. The one to use will coincide with the agent binary index actually used.

Reply:
You mean it's not right? So does that mean that `ValueParams.MirrorContentId` needs to be a `[]string` as well? Or hmm, I guess it needs to be a func that takes the stream name and returns the corresponding content id?